ext/ftp/php_ftp.c

```diff
@@ -778,8 +779,13 @@ PHP_FUNCTION(ftp_mlsd)
 
 	array_init(return_value);
 	for (ptr = llist; *ptr; ptr++) {
-		add_next_index_string(return_value, *ptr);
+		array_init(&entry);
+		// Skip invalid entry that cannot be parsed.
+		if(ftp_mlsd_parse_line(entry, *ptr) == 0) {
```

Comment:
You can add an else branch that destroys it again: `else { zval_ptr_dtor(&entry); }`